h5grove/utils.py

```diff
@@ -1,10 +1,17 @@
 import h5py
 from numbers import Number
+from os.path import basename
 import numpy as np
 from typing import Any, Sequence, Tuple, Union
 from .models import H5pyEntity
 
 
+class PathError(Exception):
+    def __init__(self, msg: str) -> None:
+        super().__init__()
+        self.msg = msg
```

Comment:
What about using `Exception.args`:
```suggestion
class PathError(Exception):
    pass
```
and to get the message:
```python
', '.join(e.args)
```

or:

```suggestion
class PathError(Exception):
    msg = property(lambda self: ', '.join(self.args))
```

Reply:
What would be the advantage ?

I see that it would indeed be more generic but as we don't need it to be generic, why bother ? :shrug:

Reply:
Not much interest indeed
I'm use to `class MyException(Exception): pass` and was surprised to see more than that.

Reply:
> I'm use to `class MyException(Exception): pass` and was surprised to see more than that.

Is it common practice to use `', '.join(e.args)` to get the error message for custom exceptions, then ?

Reply:
No, you usually don't care, but `e.args` is what you get in the API.

Actually, having a look at it, `str(e)` looks to be what you want:
```python
class PathError(Exception): pass

str(PathError("my message"))  # => "my message"
```